test/generator.hpp

```diff
@@ -33,6 +35,38 @@ namespace eve::test
       return card_t::value;
   }
 
+  //================================================================================================
+  // Poison wide data when using sub-sized types
+  //================================================================================================
+  template<simd_value W>
+  auto poison(W data)
+  {
+    using v_t = typename W::value_type;
+
+    // Add garbage at the end of sub-native registers
+    // For emulated type, there is no such gap so we don't do anything
+    if constexpr( (W::size() < eve::fundamental_cardinal_v<v_t>) && !has_emulated_abi_v<W> )
+    {
+      using p_t   = eve::as_arithmetic_t<eve::as_integer_t<v_t, unsigned>>;
+      using ftype = as_wide_t<v_t, eve::fundamental_cardinal_t<v_t>>;
+
+      ftype these(data.storage());
+
+      // Compute a recognizable filler
+      auto filler = eve::Constant<p_t, static_cast<p_t>(0xDEADBEEFC007BABE)>();
+      for(std::ptrdiff_t i=data.size();i<these.size();++i)
+      {
+        these.set(i, filler);
+      }
```

Comment:
Can we use different values?

Reply:
I'll try to mix the position of the data in the lower bits

Reply:
Done

Reply:
Is the top bit 1? So that it affects movemask?